src/components/activity-page/section.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import classNames from "classnames";
+import { Embeddable, EmbeddableImperativeAPI } from "./embeddable";
+import { isQuestion,  getLinkedPluginEmbeddable, ActivityLayouts } from "../../utilities/activity-utils";
+import { accessibilityClick } from "../../utilities/accessibility-helper";
+import IconChevronRight from "../../assets/svg-icons/icon-chevron-right.svg";
+import IconChevronLeft from "../../assets/svg-icons/icon-chevron-left.svg";
+import { EmbeddableType, SectionType } from "../../types";
+import { Logger, LogEventName } from "../../lib/logger";
+import { INavigationOptions } from "@concord-consortium/lara-interactive-api";
+
+import "./section.scss";
+
+const kEmbeddableMargin = 10;
+interface IProps {
+  section: SectionType;
+  activityLayout: number;
+  questionNumberStart: number;
+  teacherEditionMode?: boolean;
+  setNavigation?: (refId: string, options: INavigationOptions) => void;
+  pluginsLoaded: boolean;
+}
+
+export const Section: React.FC<IProps> = (props) => {
+  const { activityLayout, section, questionNumberStart } = props;
+  const [isSecondaryCollapsed, setIsSecondaryCollapsed] = useState(false);
+  const [resizeCounter, setResizeCounter] = useState(0);
+  const [primaryEmbeddableTotalHeight, setPrimaryEmbeddableTotalHeight] = useState(0);
+  const sectionDivRef = useRef<HTMLDivElement>(null);
+  const primaryDivRef = useRef<HTMLDivElement>(null);
+  const secondaryDivRef = useRef<HTMLDivElement>(null);
+  const embeddableRefs: Record<string, React.RefObject<EmbeddableImperativeAPI>> = {};
+  // used to trigger the useEffect so section can find out the height of the embeddables after they have been rendered
+  // without it, the useEffect was firing before the render was done.
+  let counter = 0;
+  const onSizeChange = useCallback(() => {
+    setResizeCounter(counter++);
+  },[counter]);
+
+  useEffect(()=>{
+    if (primaryDivRef.current !== null) {
+        if (primaryDivRef.current !== null) {
+          let totalHeight = 0;
+          for (let i=0; i<primaryDivRef.current.children.length; i++) {
+            totalHeight = totalHeight + primaryDivRef.current?.children[i].clientHeight + kEmbeddableMargin;
+          }
+          setPrimaryEmbeddableTotalHeight(totalHeight);
+      }
+    }
+  },[resizeCounter]);
+
+  const renderEmbeddables = (embeddablesToRender: EmbeddableType[], questionNumStart: number, isSingleColumn?: boolean) => {
+    let questionNumber = questionNumStart;
+    return (
+      <React.Fragment>
+        { embeddablesToRender.map((embeddable, embeddableIndex) => {
+            if (isQuestion(embeddable)) {
+              questionNumber++;
+            }
+            const linkedPluginEmbeddable = getLinkedPluginEmbeddable(section, embeddable.ref_id);
```

Comment:
I'm not sure LARA is going to put the linked plugin in the same section as the current interactive. So it would be better if `getLinkedPluginEmbeddable` was passed the `page` and then it searched the page. 

I guess that means you will need to pass the page into this this section component so it has access to it.

Reply:
Fixed this to search through the page instead.

Reply:
looks good.